src/guide/agent-functions/defining-agent-functions.rst

```diff
@@ -3,14 +3,19 @@
 Defining Agent Functions
 ^^^^^^^^^^^^^^^^^^^^^^^^
 
-Agent Functions can be specified as CUDA functions, built at compile time when using the C++ API, or they can be specified as Run-Time Compiled (RTC) function strings when using the both the C++ and Python APIs.
+Agent Functions can be specified as C++ functions, built at compile time when using the C++ API, or they can be specified as Run-Time Compiled (RTC) function strings when using the both the C++ and Python APIs. Although agent functions are technically CUDA device code (which supports only a subset of C++) the FLAME GPU API abstracts this and no CUDA syntax is required to script behaviour. Agent functions are distinguished within the examples in this guide by using the name *Agent C++*.
 
-An agent function is defined using the :c:macro:`FLAMEGPU_AGENT_FUNCTION` macro. 
+An experimental feature for Python allows the specification of *pure* agent functions in Python (referred to as *Agent Python* in this guide). The Python can then be transpiled, a process of translating the Python syntax to equivalent C++, at runtime.
+
+C++ Compile Time Agent Functions
```

Comment:
RTC is technically "compile time" too (Run time **Compiled**). 

The NVRTC docs refer to traditional non-RTC CUDA compilation as `offline static compilation`. This is being very very pedantic, so may not warrent changing but we should be careful with ambiguous language.

Reply:
Not sure I want to try and change this at this point but I get your point